include/common/core/base_types.hpp

```diff
@@ -55,6 +55,141 @@ using fp16 = sycl::half;
 ///
 using tf32 = sycl::ext::intel::experimental::esimd::tfloat32;
 
+/// @brief mx_fp4(E2M1) data packed as 8bits data type.
+struct mx_fp4 {
+  uint8_t data;
+  operator uint8_t() const {
+    return data;
+  }
+  mx_fp4() = default;
+  mx_fp4(uint8_t val) {
+    data = val;
+  }
+};
+
+template <typename T>
+struct get_packed_num {
+  static constexpr uint32_t value = 1;
+};
+
+template <>
+struct get_packed_num<mx_fp4> {
+  static constexpr uint32_t value = 2;
+};
+
+#ifdef XETLA_EMBARGO
+
+/// @brief xetla bf8 data type.
+/// The difference between bf8 and fp16 is:
+///
+/// fp16: 0_00000_0000000000
+///
+/// bf8:  0_00000_00
+/// FP32 => BF8 conversion is handled in two steps
+/// First convert to FP16, then to BF8/FP32
+/// @note
+/// The member function in bf8 class is only used in host side.
+/// For device side, we will automatically convert it to its native type.
+/// @see native_type_t
+///
+using bf8 = sycl::ext::intel::experimental::esimd::bf8;
```

Comment:
this mean dpc++ compiler supports some experimental data-type natively?
what's the dpc++ compiler version since supports these data-type?
how could we get the all experimental data-type list?

Reply:
I don't find in dpc++ compiler too, source?

Reply:
This code will be available in the future. Already deleted